FGEGraphics/GraphicsHelpers/Models/ShapeGenerators.cs

```diff
@@ -0,0 +1,254 @@
+// If neither of these are available, assume that neither you nor anyone other than the copyright holder
+// hold any right or permission to use this software until such time as the official license is identified.
+//
+
+using FGECore.ModelSystems;
+using System;
+using System.Collections.Generic;
+using OpenTK.Mathematics;
+using FGECore.CoreSystems;
+
+namespace FGEGraphics.GraphicsHelpers.Models;
+
+/// <summary>Generates shapes and returns them as a model.</summary>
+public static class ShapeGenerators
+{
+    /// <summary>
+    /// Generates a 3D sphere model.
+    /// Note that the stacks should be lower than the slices to avoid incorrect normals a good example is 10 stacks and 60 slices.
+    /// </summary>
+    public static Model GenerateSphere(float radius, uint stacks, uint slices, ModelEngine modelEngine)
+    {
+        if (stacks > slices)
+        {
+            Logs.Warning("Sphere has more stacks than slices, this may result in incorrect normals.");
+        }
+        uint vertexCount = (stacks + 1) * (slices + 1);
+        uint numIndices = stacks * slices * 6;
+        List<Vector3> vertices = new((int)vertexCount);
+        List<Vector3> normals = new((int)vertexCount);
+        List<Vector2> texCoords = new((int)vertexCount);
+        uint[] indices = new uint[numIndices];
+        int index = 0;
+        for (uint i = 0; i <= stacks; i++)
+        {
+            float theta = i * MathHelper.Pi / stacks;
+            float sinTheta = (float)Math.Sin(theta);
+            float cosTheta = (float)Math.Cos(theta);
+            for (uint j = 0; j <= slices; j++)
+            {
+                float phi = j * 2 * MathHelper.Pi / slices;
+                float sinPhi = (float)Math.Sin(phi);
+                float cosPhi = (float)Math.Cos(phi);
+                float x = radius * cosPhi * sinTheta;
+                float y = radius * cosTheta;
+                float z = radius * sinPhi * sinTheta;
+                vertices.Add(new Vector3(x, y, z));
+                normals.Add(new Vector3(x, y, z).Normalized());
+                texCoords.Add(new Vector2((float)j / slices, (float)i / stacks));
+                if (i < stacks && j < slices)
+                {
+                    uint currentRow = i * (slices + 1);
+                    uint nextRow = (i + 1) * (slices + 1);
+                    indices[index++] = currentRow + j;
+                    indices[index++] = nextRow + j;
+                    indices[index++] = currentRow + j + 1;
+                    indices[index++] = currentRow + j + 1;
+                    indices[index++] = nextRow + j;
+                    indices[index++] = nextRow + j + 1;
+                }
+            }
+        }
+        return GetModelAfterGenerating(modelEngine, "sphere", vertices, normals, texCoords, indices);
+    }
+
+    /// <summary>Generates a 2D circle model.</summary>
+    public static Model Generated2DCircle(float radius, uint slices, ModelEngine modelEngine, bool flip = false)
```

Comment:
Also stray `d` on this name

Reply:
Dyslexia strikes again! Fixed in https://github.com/FreneticLLC/FreneticGameEngine/commit/eb11317057c23bb91a2bcdc2dbcc76db965c79f7